EasyCommands/BlockHandlers/TerminalBlockHandlers.cs

```diff
@@ -94,21 +86,9 @@ public class TerminalBlockHandler<T> : BlockHandler<T> where T : class, IMyTermi
                 defaultPropertiesByPrimitive[Return.STRING] = Property.NAME;
             }
 
-            public override List<T> GetBlocksOfType(Func<IMyTerminalBlock, bool> selector) {
-                var blocks = NewList<T>();
-                PROGRAM.GridTerminalSystem.GetBlocksOfType<T>(blocks, selector);
-                return blocks;
-            }
 
-            public override List<T> GetBlocksOfTypeInGroup(String groupName) {
-                var blockGroups = NewList<IMyBlockGroup>();
-                PROGRAM.GridTerminalSystem.GetBlockGroups(blockGroups);
-                IMyBlockGroup group = blockGroups.Find(g => g.Name.Equals(groupName));
-                var blocks = NewList<T>();
-                if (group == null) return blocks;
-                group.GetBlocksOfType<T>(blocks);
-                return blocks;
-            }
+            public override IEnumerable<T> SelectBlocksByType<U>(List<U> blocks, Func<U, bool> selector = null) =>
+                blocks?.Where(b => selector?.Invoke(b) ?? true).OfType<T>() ?? Enumerable.Empty<T>();
```

Comment:
Do we need the null check against blocks and the fallback ENumerable.Empty<T>()?  I don't think we ever call this method with null blocks, do we?

Reply:
maybe i was a bit paranoid there ...